src/axom/core/utilities/System.cpp

```diff
@@ -97,7 +97,7 @@ std::locale locale(const std::string& name)
   }
   catch(std::runtime_error&)
   {
-    loc = std::locale(loc, "", std::locale::ctype);
+    loc = std::locale(loc, "C", std::locale::ctype);
```

Comment:
Locales are the gift that keep on giving!

The standard appears to require `std::locale("C")` and `std::locale("")`, but apparently the latter cannot always be relied upon. See: https://stackoverflow.com/a/46364980

The original was failing in our AWS docker image (but passing in our docker-based CI, and everywhere else I've run it).

See also: https://github.com/LLNL/axom/issues/1128